File: allure2-androidj/src/main/java/com/yq/allure2_androidj/common/resultRW/gson/SerializationProcessor.java
package com.yq.allure2_androidj.common.resultRW.gson;

import java.io.File;

public interface SerializationProcessor extends Serializer, Deserializer{
}

interface Serializer {
    String serialize(Object var1);

    void serialize(File var1, Object var2);
}

interface Deserializer {
    <T> T deserialize(String var1, Class<T> var2);

    <T> T deserialize(File var1, Class<T> var2);
}

File: allure2-android/src/main/java/com/yq/allure2_android/common/serialization/SerializationProcessor.kt
package com.yq.allure2_android.common.serialization

import java.io.File

/**
 * @author king
 */
interface SerializationProcessor : Serializer, Deserializer

interface Serializer {
    fun serialize(src: Any): String
    fun serialize(file: File, src: Any)
}

interface Deserializer {
    fun <T> deserialize(string: String, type: Class<T>): T
    fun <T> deserialize(file: File, type: Class<T>): T
}

